### PR TITLE
feat: transaction history table

### PR DESCRIPTION
**Pull request by jpangelle**
### Description

This pull request implements the transaction history table to the pool detail page redesign.

#1891

### Approvals

- [x] Dev
- [x] Designer
- [x] Product

### Screenshots

image

### Impact

Pool detail page

### CODE REVIEW

File: centrifuge-js/src/modules/pools.ts [hunk 1]
```diff
@@ -2933,11 +2939,11 @@ export function getPoolsModule(inst: Centrifuge) {
...
-          //   const { timestamp, value } = oracle[1].toPrimitive() as any
-          //   oraclePrices[(oracle[0].toHuman() as any)[0].Isin] = {
-          //     timestamp,
-          //     value: new CurrencyBalance(value, currency.decimals),
-          //   }
+        //   const { timestamp, value } = oracle[1].toPrimitive() as any
+        //   oraclePrices[(oracle[0].toHuman() as any)[0].Isin] = {
+        //     timestamp,
+        //     value: new CurrencyBalance(value, currency.decimals),
+        //   }
```
Comment (jpangelle) [inline centrifuge-js/src/modules/pools.ts:2946]: `prettier` formatting

File: centrifuge-app/src/pages/PoolTransactions.tsx
Comment (jpangelle) [inline centrifuge-app/src/pages/PoolTransactions.tsx:1]: New page `/pools/:pid/transactions`

Comment (jpangelle) [inline centrifuge-app/src/pages/PoolTransactions.tsx:1]: New page `/pools/:pid/transactions`

File: centrifuge-js/src/modules/pools.ts [hunk 2]
```diff
@@ -10,22 +10,22 @@ import { SolverResult, calculateOptimalSolution } from '..'
...
-    getDateMonthsFromNow,
-    getDateYearsFromNow,
-    getRandomUint,
-    isSameAddress,
+  addressToHex,
+  computeTrancheId,
+  getDateMonthsFromNow,
+  getDateYearsFromNow,
+  getRandomUint,
+  isSameAddress,
```
Comment (jpangelle) [inline centrifuge-js/src/modules/pools.ts:28]: `prettier` formatting

File: centrifuge-app/.env-config/.env.development
Comment (jpangelle) [inline centrifuge-app/.env-config/.env.development:1]: will revert before merging

File: centrifuge-app/src/components/PoolOverview/TransactionHistory.tsx [hunk 3]
```diff
@@ -1,9 +1,179 @@
...
+}
+
+const getTransactionTypeStatus = (type: string) => {
+  if (type === 'Principal payment' || type === 'Repaid') return 'warning'
+  if (type === 'Interest') return 'ok'
+  return 'default'
+}
+
+const getAmount = (amount: string | CurrencyBalance | null) => {
+  if (typeof amount === 'string') return formatBalance(new CurrencyBalance(amount, 6), 'USD', 2, 2)
```
[the remark this answers is not in the document]
Comment (jpangelle) [reply centrifuge-app/src/components/PoolOverview/TransactionHistory.tsx]: resolved 😄

**Review by sophialittlejohn — APPROVED** [review]
🔥

**Pull request merged**